behind-the-veil-siteroot/imports/ui/components/pages/login/LoginPage.jsx

```diff
@@ -19,34 +19,61 @@ import UrlBasePath from "../../../enums/UrlBasePath";
  * The Sign-In Page for the website.
  */
 export const LoginPage = () => {
-
     // Constant navigate will allow the user to be redirected to different webpages.
     const navigate = useNavigate();
 
+    // form inputs and errors
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [errors, setErrors] = useState({
+        username: "",
+        password: ""
+    });
+
+    // get whether it errored before in the URL
+    const urlParams = new URLSearchParams(window.location.search);
+
+    // if checks that display error is updated, prevents infinite loop
+    if (urlParams.get("error") === "true" && !errors.password) {
+        const newErrors = {}
+        newErrors.password = "Username or password incorrect";
+        setErrors(newErrors)
```

Comment:
Not sure whether this was intentional, but wouldn't this set errors to be an object with just "password" as a key without "username" as a key?

Reply:
yep thats correct, only the errors that are there will be set